pallets/dkg-proposal-handler/src/lib.rs

```diff
@@ -831,6 +948,74 @@ impl<T: Config> Pallet<T> {
 		Ok(header)
 	}
 
+	fn decode_rescue_tokens_proposal(data: &[u8]) -> Result<ProposalHeader<T::ChainId>, Error<T>> {
+		if data.len() != 112 {
+			return Err(Error::<T>::ProposalFormatInvalid)?
+		}
+		let header = Self::decode_proposal_header(data)?;
+		let mut token_address_bytes = [0u8; 20];
+		token_address_bytes.copy_from_slice(&data[40..60]);
+		let token_address = Address::from(token_address_bytes);
+		let mut to_bytes = [0u8; 20];
+		to_bytes.copy_from_slice(&data[60..80]);
+		let to = Address::from(to_bytes);
+		let mut amount_to_rescue_bytes = [0u8; 32];
+		amount_to_rescue_bytes.copy_from_slice(&data[80..112]);
+		Ok(header)
+	}
+
+	/// (header: 40 Bytes, max_deposit_limit_bytes: 32) = 72
+	/// Bytes
+	fn decode_max_deposit_limit_proposal(
```

Comment:
Functions look great and they all do the same thing, can we aggregate the logic in one function for parameters with this byte lengths? Maybe using const generics or do you think it's a bad idea to de-deduplicate the code?

Reply:
Yh... I was thinking of making it a one function call for all but then I thought what if the byte length for each of them changes in the future, well I guess by then we can make the decoding specific. But, for now, I will make the function just one for all as you suggested.